src/jdk.jlink/share/classes/jdk/tools/jmod/JmodTask.java

```diff
@@ -1179,6 +1181,30 @@ public LocalDateTime convert(String value) {
         @Override public String valuePattern() { return "date"; }
     }
 
+    static class CompLevelConverter implements ValueConverter<Integer> {
+        @Override
+        public Integer convert(String value) {
+            int idx = value.indexOf("-");
+            int lastIdx = value.lastIndexOf("-");
+            if (idx != lastIdx) {
+                throw new CommandException("err.compress.incorrect", value);
+            }
```

Comment:
what about checking `value.substring(0, idx)` is "zip"?

Reply:
Right, missed that! Fixed in new commit.